packages/product/src/services/product-module-service.ts

```diff
@@ -649,19 +654,22 @@ export default class ProductModuleService<
         productVariantsMap.set(productData.handle!, variants ?? [])
         productOptionsMap.set(productData.handle!, options ?? [])
 
-        if (!productData.thumbnail && productData.images?.length) {
```

Comment:
the thumbnail code is not needed?

Reply:
No you have encapsulated it in another method that manages the images